code/controllers/subsystem/processing/networks.dm

```diff
@@ -1,51 +1,135 @@
+#define DEBUG_NETWORKS
+
 PROCESSING_SUBSYSTEM_DEF(networks)
 	name = "Networks"
 	priority = FIRE_PRIORITY_NETWORKS
 	wait = 1
 	stat_tag = "NET"
 	flags = SS_KEEP_TIMING
 	init_order = INIT_ORDER_NETWORKS
+
 	var/datum/ntnet/station/station_network
-	var/assignment_hardware_id = HID_RESTRICTED_END
-	var/list/networks_by_id = list()				//id = network
-	var/list/interfaces_by_id = list()				//hardware id = component interface
-	var/resolve_collisions = TRUE
+	var/list/network_initialize_queue = list()
+
+	var/list/interfaces_by_hardware_id = list()
+	var/list/networks = list()
+	// Used with map tags to look up hardware address
+
+/datum/controller/subsystem/processing/stat_entry(msg)
+	msg = "[stat_tag]:[length(processing)]"
+	return ..()
+
 
 /datum/controller/subsystem/processing/networks/Initialize()
 	station_network = new
-	station_network.register_map_supremecy()
-	. = ..()
 
-/datum/controller/subsystem/processing/networks/proc/register_network(datum/ntnet/network)
-	if(!networks_by_id[network.network_id])
-		networks_by_id[network.network_id] = network
-		return TRUE
-	return FALSE
+	return ..()
+
+/datum/controller/subsystem/processing/networks/fire(resumed = 0)
+	// so life sucks.  Can't be in Initialize because Initialize is run async and we must start
```

Comment:
What are you talking about, subsystems have strict initialization order.

Reply:
Doesn't.  atmosinit() runs during mapload, async during the mapping process BEFORE Initialize.  On each template.  So yes, while Initialize does load after that map has loaded it screwed me up because vents and scrubbers soft'ed Initialized before that.  I refractored a lot of atmos and now I am getting rid of that subsystem hack so we just need to use Initialize.  It just took to much digging into how the atmos.dm subsystem worked and pissed me off.